src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/JWKKeyAccessor.kt

```diff
@@ -0,0 +1,86 @@
+
+@Component
+class JWKKeyAccessor(
+  @Value("\${jwt.signing.key.pair}") private val privateKeyPair: String,
+  @Value("\${jwt.keystore.password}") private val keystorePassword: String,
+  @Value("\${jwt.keystore.alias:elite2api}") private val keystoreAlias: String,
+  @Value("\${jwt.jwk.key.id}") private val keyId: String,
+  @Value("#{'\${jwt.auxiliary.keystore.alias}' <= '' ? null : '\${jwt.auxiliary.keystore.alias}'}") private val keystoreAliasAuxiliary: String?,
+  @Value("#{'\${jwt.auxiliary.keystore.password}' <= '' ? null : '\${jwt.auxiliary.keystore.password}'}") private val keystorePasswordAuxiliary: String?,
+  @Value("#{'\${jwt.auxiliary.signing.key.pair}' <= '' ? null : '\${jwt.auxiliary.signing.key.pair}'}") private val privateKeyPairAuxiliary: String?,
+  @Value("#{'\${jwt.auxiliary.jwk.key.id}' <= '' ? null : '\${jwt.auxiliary.jwk.key.id}'}") private val keyIdAuxiliary: String?,
+) {
+  private val primaryKeyStore: KeyStore = initializePrimaryKeyStore()
+  private val auxiliaryKeyStore: KeyStore = initializeAuxiliaryKeyStore()
+  fun getPrimaryKeyPair(): KeyPair {
+    val privateKey = primaryKeyStore.getKey(keystoreAlias, keystorePassword.toCharArray()) as? RSAPrivateCrtKey
+    val rsaPublicKeySpec = RSAPublicKeySpec(privateKey!!.modulus, privateKey.publicExponent)
+    val publicKey = KeyFactory.getInstance("RSA").generatePublic(rsaPublicKeySpec)
+    return KeyPair(publicKey, privateKey)
+  }
+  fun getAuxiliaryKeyPair(): KeyPair? {
+    return if (ObjectUtils.anyNull(keystoreAliasAuxiliary, keystorePasswordAuxiliary, keyIdAuxiliary)) {
+      null
+    } else {
+      val privateKey =
+        auxiliaryKeyStore.getKey(keystoreAliasAuxiliary, keystorePasswordAuxiliary!!.toCharArray()) as RSAPrivateCrtKey
+      val rsaPublicKeySpec = RSAPublicKeySpec(privateKey.modulus, privateKey.publicExponent)
+      val publicKey = KeyFactory.getInstance("RSA").generatePublic(rsaPublicKeySpec)
+      return KeyPair(publicKey, privateKey)
+    }
+  }
+
+  fun getAuxiliaryPublicKey(): RSAKey? {
+    return if (ObjectUtils.anyNull(keyIdAuxiliary, keystoreAliasAuxiliary, keystorePasswordAuxiliary, privateKeyPairAuxiliary)) {
+      null
+    } else {
+      return return buildRSAKey(getAuxiliaryKeyPair(), keyIdAuxiliary)
+    }
+  }
+
+  fun getPrimaryPublicKey(): RSAKey {
+    return buildRSAKey(getPrimaryKeyPair(), keyId)
+  }
+  private fun initializePrimaryKeyStore(): KeyStore {
+    val store = KeyStore.getInstance("jks")
+    val primaryStoreInputStream = ByteArrayResource(decodeBase64(privateKeyPair)).inputStream
+
+    primaryStoreInputStream.use {
+      store.load(it, keystorePassword.toCharArray())
+    }
+    return store
+  }
+
+  private fun buildRSAKey(keyPair: KeyPair?, keyId: String?): RSAKey {
+    return keyPair.let { RSAKey.Builder(it?.public as? RSAPublicKey) }
+      .keyUse(KeyUse.SIGNATURE)
+      .algorithm(JWSAlgorithm.RS256)
+      .keyID(keyId).build()
+  }
+  private fun initializeAuxiliaryKeyStore(): KeyStore {
```

Comment:
Is there a reason why the implementation of this function is different to `initializePrimaryKeyStore`?

Reply:
Updated